src/k2/module/k23si/Module.cpp

```diff
@@ -1108,11 +1108,40 @@ K23SIPartitionModule::_processWrite(dto::K23SIWriteRequest&& request, FastDeadli
         }
     }
 
+    // build write_ranges for Transaction end request
+    std::unordered_map<String, std::unordered_set<dto::KeyRangeVersion>> write_ranges; // will be filled in later if WI is created successfully
+
+    //endRequest created for calling EndTransaction for one hot transactions.
+    dto::K23SITxnEndRequest endRequest{request.pvid,
+        request.trhCollection,
+        request.trh,
+        request.mtr,
+        dto::EndAction::Commit,
+        std::move(write_ranges)
+    };
+
+    return seastar::do_with((dto::K23SITxnEndRequest)std::move(endRequest), (bool)request.isonehot,
```

Comment:
no need for do-with. just move-capture in the lambda

Reply:
done